x-pack/test/reporting_api_integration/reporting_without_security/job_apis_csv.ts

```diff
@@ -54,31 +45,21 @@ export default function ({ getService }: FtrProviderContext) {
       await esArchiver.load('x-pack/test/functional/es_archives/logstash_functional');
     });
 
+    afterEach(async () => {
+      await reportingAPI.deleteAllReports();
+    });
+
     after(async () => {
       await reportingAPI.teardownLogs();
       await esArchiver.unload('x-pack/test/functional/es_archives/logstash_functional');
     });
 
-    afterEach(async () => {
-      await reportingAPI.deleteAllReports();
-    });
-
     it('Posted CSV job is visible in the job count', async () => {
       const { job, path } = await postJobCSV();
-      expectSnapshot(pick(job, apiResponseFields)).toMatchInline(`
-        Object {
-          "attempts": 0,
-          "created_by": false,
-          "jobtype": "csv_searchsource",
-          "meta": Object {
-            "objectType": "search",
-          },
-          "payload": Object {
-            "title": "A Saved Search With a DATE FILTER",
-          },
-          "status": "pending",
-        }
-      `);
+      expect(job.payload.title).equal('A Saved Search With a DATE FILTER');
+
+      // wait for index refresh
+      await sleep(3000);
```

Comment:
I wonder how risky this is to introduce flakiness. Wouldn't it be better to pass a param to `postJobCSV({ refreshIndex: true })` and avoid the `sleep` call?

Reply:
Good point!
  * Passing many test runs with the flaky test runner is definitely a requirement for this PR.
  * Calling for a refresh on the index whenever a job is posted seems expensive enough to merit seconds thoughts. 

The amount of time to sleep should be derived from the refresh interval for data streams, which I think is 3 seconds (this is integration test is for stateful). However, in serverless it is 15 seconds. I will confirm these things in a sync with @elastic/es-data-management

Reply:
(disclaimer about not really understanding the typescript code)

Could this use the `wait_for` option when indexing things so that you only wait for as long as is necessary for the refresh to happen naturally? I.e.: https://www.elastic.co/guide/en/elasticsearch/reference/8.12/docs-refresh.html#docs-refresh

Reply:
@sebelga I think I understand now. But, I'm very hesitant to add a flag in the API that would only serve the purpose of tests.

@dakrone I will try your suggestion to use `wait_for`, and see if that allows for removing or lowering the sleep call.

If the sleep is needed, my goal will to make sure it is set to a calculated amount of time. It should be low enough to wait for ES and avoiding an understood race condition. It shouldn't be perceived as something to address flakiness.

Reply:
I used the suggestion to change to `refresh=wait_for` in the operations to index new documents and update documents, and it seems to resolved this.